packages/yarnpkg-core/sources/Cache.ts

```diff
@@ -184,11 +184,11 @@ export class Cache {
       return zipFs;
     };
 
-    const validateFile = async (path: PortablePath, refetchPath: PortablePath | null = null) => {
+    const validateFile = async (path: PortablePath, refetchPath: PortablePath | null = null): Promise<[isValid: boolean, hash: string | null]> => {
```

Comment:
Please return an object instead of a tuple for this, it makes it easier when reading the code to see what's going on.

Reply:
fixed